daphne_worker/src/dap.rs

```diff
@@ -64,29 +65,88 @@ pub(crate) fn dap_response_to_worker(resp: DapResponse) -> Result<Response> {
     Ok(worker_resp)
 }
 
+#[async_trait(?Send)]
 impl<D> HpkeDecrypter for DaphneWorkerConfig<D> {
-    fn get_hpke_config_for(&self, _task_id: &Id) -> Option<&HpkeConfig> {
-        if self.hpke_receiver_config_list.is_empty() {
-            return None;
-        }
+    async fn get_hpke_config_for(
+        &self,
+        _task_id: &Id,
+    ) -> std::result::Result<Option<HpkeConfig>, DapError> {
+        let kv_store = self.kv("DAP_HPKE_RECEIVER_CONFIG_STORE").map_err(dap_err)?;
+        let num_kv_keys: u64 = 1; // Used to limit the number of keys we retreive from KV.
+        let keys = kv_store
+            .list()
+            .limit(num_kv_keys)
+            .prefix(KV_KEY_PREFIX_HPKE_RECEIVER_CONFIG.to_string())
+            .execute()
+            .await
+            .map_err(|e| DapError::Fatal(e.to_string()))?;
```

Comment:
```suggestion
            .map_err(dap_err)?;
```

Reply:
This doesn't work, because `dap_err` converts `worker::Error` into `daphne::DapError`.

Reply:
Isn't that exactly what we want? What error type does `execute()` return? I would have thought it would be `worker::Error`.

Reply:
Oh sorry, I forgot to mention, but `execute()` returns `KvError`, not `worker::Error`.

Reply:
The latest push should address this.

Reply:
good catch